### PR TITLE
 config: avoid some buffer allocs

### PR DESCRIPTION
**Pull request by euank**
strings.NewReader just uses a couple ints to keep track of where it's
reading whereas the bytes variant copies all of the contents of the
string.

This is a microoptimization, but I think it's also mildly cleaner code.

I double checked my understanding that strings.NewReader doesn't allocate so much in the [playground](https://play.golang.org/p/ncF4-yYei0h). In reality, units are pretty small so the difference will be minuscule.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.PHONY: build
```
Comment (lucab) [inline Makefile]: I'm confused, shouldn't phony be specific to the target (`all`) instead of the filename (`build`)?

Comment (euank) [reply Makefile]: Oops, good catch! I meant to name it `build` instead of `all`, silly typo.

**Comment by euank** [comment]
Updated, addressed comment

**Comment by dgonyeo** [comment]
@euank if you force-push tailor should re-run, and then this can get merged.

**Comment by euank** [comment]
Whoops, didn't notice we were waiting on tailor. no-op amended

**Comment by dgonyeo** [comment]
good to merge when you are

File: Makefile (hunk 1 above)
Comment (lucab) [reply Makefile]: This was my saddest review ever, I was honestly expecting to learn some new makefile trick :(

**Pull request merged**